src/js/popup/popup.js

```diff
@@ -294,21 +259,30 @@ async function showRelayPanel(tipPanelToShow) {
   const currentPanel = onboardingPanelWrapper.querySelector(".current-panel");
   const upgradeButtonEl = onboardingPanelWrapper.querySelector(".upgrade-banner");
   const upgradeButtonIconEl = onboardingPanelWrapper.querySelector(".upgrade-banner-icon");
-  const panelPagination = onboardingPanelWrapper.querySelector(".onboarding-pagination");
-  const onboardingPanelStrings = getOnboardingPanels();
-  const educationalStrings = getEducationalStrings();
+  const promoElements = onboardingPanelWrapper.querySelectorAll(".js-new-label");
+  const tipCtaEl = onboardingPanelWrapper.querySelector(".onboarding-cta");
+  let premiumPanelStrings = getEducationalStrings();
+  let onboardingPanelStrings = await getOnboardingPanels();
 
-  document.querySelectorAll(".total-panels").forEach(panel => {
-    panel.textContent = 2;
-  });
+  const isBundleAvailableInCountry = (await browser.storage.local.get("bundlePlans")).bundlePlans.BUNDLE_PLANS.available_in_country;
+  const isPhoneAvailableInCountry = (await browser.storage.local.get("phonePlans")).phonePlans.PHONE_PLANS.available_in_country;
+  
+  // If Bundle & Phone flags are enabled, show the promo announcements and promo elements, else hide it
+  if (checkWaffleFlag("bundle") && checkWaffleFlag("phones")
+   && isBundleAvailableInCountry && isPhoneAvailableInCountry
+  ) {
+    promoElements.forEach(i => {
+      i.style.display = "block";
+    });
+    onboardingPanelWrapper.setAttribute("id", "bundle-phones-promo");
+    onboardingPanelStrings = await getPromoPanels();
+    premiumPanelStrings = await getPromoPanels();
+  }
```

Comment:
This is the logic for switching out previous announcement content for the Bundle + Phone masking contents if all of the conditions are met.